### PR TITLE
Add p3a UI android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/12723

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
For new users and existing users please verify the selection in settings->privacy->Automatically send completely private product analytics to Brave option. if the checkbox is selected, this option should be selected and viceversa

### CODE REVIEW

**Review by samartnik — APPROVED** [review]
lgtm

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -308,21 +310,17 @@ public void finishNativeInitialization() {
...
-            // } else {
-            //     // On the first run P3A is must be set from the onboarding wizard
-            // }
-            // Uncomment ^ when onboarding UI step will be ready for P3A
-
-            // Until we don't have P3A onboarding, P3A will be disabled,
-            // but available for activating through settings
             if (!BravePrefServiceBridge.getInstance().hasPathP3AEnabled()) {
                 BravePrefServiceBridge.getInstance().setP3AEnabled(false);
             }
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:320]: do we need that duplicated code here? we have it above as well before the condition check https://github.com/brave/brave-core/pull/7550/files#diff-448df90d6aa5a20513ca5dba42a4e428dfd22f05577b904f8fe7986c40ab90faR313

File: android/java/org/chromium/chrome/browser/onboarding/P3aOnboardingActivity.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/onboarding/P3aOnboardingActivity.java:2]: Year 2021 :+1:

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 3]
```diff
@@ -309,20 +311,14 @@ public void finishNativeInitialization() {
         BraveSyncReflectionUtils.showInformers();
 
         if (BraveConfig.P3A_ENABLED) {
-            // // P3A informer should be shown not for the updated users only
-            // if (!PackageUtils.isFirstInstall(this)) {
-            //     BraveP3AInformers.show();
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:314]: PR looks good :+1: 

The only notice is to remove `brave/android/java/org/chromium/chrome/browser/informers/BraveP3AInformers.java`, as it is not used anymore, and it's functions are implemented in `brave/android/java/org/chromium/chrome/browser/onboarding/P3aOnboardingActivity.java`

**Review by AlexeyBarabash — APPROVED** [review]
LGTM :+1:

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**